arvo/clay/examples.md

```diff
@@ -0,0 +1,841 @@
+
+`sub-mult.hoon`
+
+```hoon
+/-  spider 
+/+  strandio
+=,  strand=strand:spider 
+^-  thread:spider 
+|=  arg=vase 
+=/  m  (strand ,vase) 
+^-  form:m
+=/  files=(set (pair care:clay path))
+  %-  sy  :~
+            [%x /foo/txt]
+            [%u /bar/txt]
+          ==
+;<  =bowl:strand  bind:m  get-bowl:strandio
+=/  =task:clay  [%warp our.bowl %home ~ %mult da+now.bowl files]
+=/  =card:agent:gall  [%pass /mult %arvo %c task]
+;<  ~  bind:m  (send-raw-card:strandio card)
+;<  response=(pair wire sign-arvo)  bind:m  take-sign-arvo:strandio
+~&  +.q.response
+(pure:m !>(~))
+```
+
+Save the above to `ted/sub-mult.hoon`, `|commit %home` and run with `-sub-mult`. Now, create `foo.txt` and `bar.txt` in your home directory, hit backspace in the dojo to disconnect the thread and run `|commit %home`. You should see something like:
+
+```
+> |commit %home
+>=
+[%wris p=[%da p=~2021.4.27..06.07.08..5ec4] q={[p=%u q=/bar/txt] [p=%x q=/foo/txt]}]
++ /~zod/home/151/foo/txt
++ /~zod/home/151/bar/txt
+```
+
+You'll notice that, unlike a `%writ`, the `%wris` doesn't give you the data. It merely tells you the `care`s and `path`s of the files that changed. If you need to actually get the data, you can just scry or send a request for the files in question.
+
+Now, run the thread again, open `bar.txt` in an editor, modify its contents, save it and `|commit %home`. You'll notice you didn't receive a `%wris`. This is because we subscribed to `/bar/txt` with `%u` care and its existence didn't change.
+
+Lastly, delete `foo.txt` and `|commit %home`. You should see something like:
+
+```
+> |commit %home
+>=
+[%wris p=[%da p=~2021.4.27..06.15.03..0da4] q={[p=%x q=/foo/txt]}]
+- /~zod/home/153/foo/txt
+```
+
+As you can see, a relevant change to any of the subscribed files will trigger a response, not just when all of them change.
+
+Here's a breakdown of the `task` we sent:
+
+![subscribe mult diagram](https://pub.m.tinnus-napbus.xyz/sub-mult.png "subscribe mult diagram")
+
+## %many
+
+Here we'll look at subscribing to a range of changes to a `desk` by passing Clay a `%warp` `task` with a `%many` `rave` and receiving `%writ` `gift`s when changes occur.
+
+This thread will subscribe to changes to your `%home` `desk` for the next three minutes. The `track` is `%.y` so it will only inform you of changes, not send the full `nako`. It will only get updates if the specified file exists. It contains a `main-loop` that will take an arbitrary number of `sign`s and print them out in the dojo. Since it never ends, you'll need to stop it with the `:spider|kill` command in the dojo.
+
+`sub-many.hoon`
+
+```hoon
+/-  spider 
+/+  strandio
+=,  strand=strand:spider
+|%
+++  take-sign-loop
+  =/  m  (strand ,~)
+  ^-  form:m
+  %-  (main-loop:strandio ,~)
+  :~  |=  ~
+      ^-  form:m
+      ;<  res=(pair wire sign-arvo)
+        bind:m
+      ((handle:strandio ,(pair wire sign-arvo)) take-sign-arvo:strandio)
```

Comment:
```suggestion
      ;<    res=(pair wire sign-arvo)
          bind:m
        ((handle:strandio ,(pair wire sign-arvo)) take-sign-arvo:strandio)
```

Should be indented like this so only the last child (the ~& and pure) are at the same level as the initial rune